docs/data/data-grid/aggregation/aggregation.md

```diff
@@ -4,15 +4,277 @@ title: Data Grid - Aggregation
+isGroupAggregated={(groupNode) => groupNode?.groupingField === 'company'}
+
+// Will only aggregate the company group "Universal Pictures"
+isGroupAggregated={(groupNode) =>
+  groupNode?.groupingField === 'company' &&
+  groupNode?.groupingKey === 'Universal Pictures'
+}
+
+// Will aggregate on the grouping rows and on the top level footee
+isGroupAggregated={(groupNode, position) => position === 'inline' || groupNode == null}
+```
+
+The demo below shows the _sum_ aggregation on both the grouping rows and the top level footer.
+
+{{"demo": "AggregationIsGroupAggregated.js", "bg": "inline"}}
+
+## Usage with tree data
+
+Like for row grouping, you can display the aggregated values either in the footer or in the grouping row.
+
+:::info
+If the aggregated value is displayed in the grouping row, it will always have priority over the row data.
+This means that the data from groups explicitly provided in your dataset will be ignored in favour of their aggregated values.  
+:::
+
+The demo below shows the _sum_ aggregation on the "Size" column and the _max_ aggregation on the "Last modification" column.
+
+{{"demo": "AggregationTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Filtering
+
+By default, the aggregation only uses the filtered rows.
+You can set the `aggregatedRows` to `"all"` to use all rows.
+
+In the example below, the movie _Avatar_ is not passing the filters but is still used for the **max** aggregation of the `gross` column.
+
+{{"demo": "AggregationFiltering.js", "bg": "inline"}}
+
+## Aggregation functions
+
+### Basic structure
+
+An aggregation function is an object describing how to generate an aggregated value for a given set of values.
+
+Let's take a look at a simple aggregation function:
+
+The full typing details can be found on the [GridAggregationFunction API page](/x/api/data-grid/grid-aggregation-function/).
+
+### Built-in functions
+
+The `@mui/x-data-grid-premium` package comes with a set of built-in aggregation functions to cover the basic use-cases:
+
+| Name   | Behavior                                                   | Supported column types       |
+| ------ | ---------------------------------------------------------- | ---------------------------- |
+| `sum`  | Returns the sum of all values in the group                 | `number`                     |
+| `avg`  | Returns the non-rounded average of all values in the group | `number`                     |
+| `min`  | Returns the smallest value of the group                    | `number`, `date`, `dateTime` |
+| `max`  | Returns the largest value of the group                     | `number`, `date`, `dateTime` |
+| `size` | Returns the amount of cells in the group                   | all                          |
+
+### Remove a built-in function
+
+#### Remove a built-in function for all columns
+
+You can remove some aggregation functions for all columns by passing a filtered object to the `aggregationFunctions` prop.
```

Comment:
Could the users do something like `aggregationFunctions:{[GRID_AGGREGATION_FUNCTIONS.min, GRID_AGGREGATION_FUNCTIONS.max]}` ?

If yes, perhaps we should set it as an "entry" example (inline).

Reply:
No, the format is an object, not an array

```tsx
aggregationFunctions:{{
  min: GRID_AGGREGATION_FUNCTIONS.min, 
  max: GRID_AGGREGATION_FUNCTIONS.max
}}
```

Reply:
@joserodolfofreitas